src/Compilers/CSharp/Portable/SymbolDisplay/SymbolDisplayVisitor.cs

```diff
@@ -93,8 +93,21 @@ protected override void FreeNotFirstVisitor(AbstractSymbolDisplayVisitor visitor
 
         internal SymbolDisplayPart CreatePart(SymbolDisplayPartKind kind, ISymbol? symbol, string text)
         {
-            text = (text == null) ? "?" :
-                   (_escapeKeywordIdentifiers && IsEscapable(kind)) ? EscapeIdentifier(text) : text;
+            if (text == null)
+            {
+                return new SymbolDisplayPart(kind, symbol, "?");
+            }
+
+            if (!_escapeKeywordIdentifiers)
+            {
+                return new SymbolDisplayPart(kind, symbol, text);
+            }
+
+            bool isNamedTypeOrAliasName = symbol?.Kind is SymbolKind.NamedType or SymbolKind.Alias;
+
+            bool isEscapable = IsEscapable(kind);
+
+            text = isEscapable ? EscapeIdentifier(text, isNamedTypeOrAliasName) : text;
```

Comment:
Consider rearranging to avoid calculating `isNamedTypeOrAliasName` or reassigning `text` unnecessarily. Perhaps:
```csharp
if (_escapeKeywordIdentifiers && IsEscapable(kind))
{
    text = EscapeIdentifer(text, symbol?.Kind is SymbolKind.NamedType or SymbolKind.Alias);
}
```

Reply:
Actually, looking at the previous line, we return if `!escapeKeywordIdentifer` - so we can omit that check here.

Reply:
Yes, we're checking `!escapeKeywordIdentifiers` earlier but the two `if` statements could be combined to avoid two `return new SymbolDisplayPart(kind, symbol, text);`.